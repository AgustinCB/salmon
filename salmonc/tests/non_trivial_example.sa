class TestModule {
    var variable = 1;

    init() {
        variable = variable + 1;
    }

    test() {
        print "hola";
    }

    test1() {
        print variable;
    }
}

trait TestTrait {
    echo(a);
}

class TestClass < TestModule {
    init() {
        this.super.init();
    }
    testmethod() {
        print "hola method";
        print this.super.variable;
    }
}

trait MyTrait {
    mytraitmethod();
}

class Impl < TestClass {}

trait MyTrait for TestClass {
    mytraitmethod() {
        print "my trait method";
    }
}

trait TestTrait for Impl {
  echo(a) {
      print a;
  }
}
var test = TestModule();

test.test();

fun counter() {
    class CounterClass {
        var counter = 0;
        addone() {
            counter = counter + 1;
        }
    }
    return CounterClass();
}

var c = TestClass();
c.testmethod();
c.mytraitmethod();

var i = Impl();
i.echo("pepe");
i.super.testmethod();
i.testmethod();

print test.variable;
test.test1();

var c1 = counter();
print c1.counter;
c1.addone();
print c1.counter;
