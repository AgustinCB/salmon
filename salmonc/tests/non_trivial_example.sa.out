hola
hola method
2
my trait method
pepe
hola method
3
hola method
1
2
2
0
1
